src/test/scala/sigmastate/lang/SigmaSpecializerTest.scala

```diff
@@ -77,4 +77,15 @@ class SigmaSpecializerTest extends PropSpec with PropertyChecks with Matchers wi
         Where(Outputs, 21, GE(ExtractAmount(TaggedBox(21)), LongConstant(10)))
   }
 
+  property("AND flattening") {
+    spec("true && true && true") shouldBe AND(TrueLeaf, TrueLeaf, TrueLeaf)
+    spec("true && (true && (true && true)) && true") shouldBe
+      AND(TrueLeaf, TrueLeaf, TrueLeaf, TrueLeaf, TrueLeaf)
+  }
+
+  property("OR flattening") {
+    spec("true || true || true") shouldBe OR(TrueLeaf, TrueLeaf, TrueLeaf)
+    spec("true || (true || true) || true") shouldBe
+      OR(TrueLeaf, TrueLeaf, TrueLeaf, TrueLeaf)
+  }
 }
```

Comment:
@greenhat please add more comprehensive test cases
true && true
true && false
true && (true && 10 == 10)
(10 == 10 && 10 == 10 && 10 == 10)
etc. all different possible combinations

Reply:
Thanks! Will do. If only I could figure out a way to employ prop checks here. No worthy ideas so far but I'll keep digging.